content/04-git-command-line/12-pull-upstream-changes.en.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Pulling Upstream Changes
+weight: 12
+---
+
+Now that everyone has contributed to the [**RSE-Sheffield** repository](https://github.com/RSE-Sheffield/collaborative_github_exercise) (the **upstream** repository), the `params/` folder contains a number of files, one for each successful pull request.
+
+However your local copy of the repository only contains the template and your own file.
+
+> Q: How can I merge changes from the **upstream** repository to my **local** repository?
+
+
+### Add the **Upstream** Remote repository
+
+First, we have to configure `git` to know where our **upstream** repository is located by running
+```
+$ git remote add upstream https://github.com/RSE-Sheffield/collaborative_github_exercise.git
+```
+to add a remote repository named `upstream` at the URL `https://github.com/RSE-Sheffield/collaborative_github_exercise.git`.
+We can check that this worked using the following
+```
+$ git remote -v
+
+origin	git@github.com:johndoe/collaborative_github_exercise.git (fetch)
+origin	git@github.com:johndoe/collaborative_github_exercise.git (push)
+upstream	git@github.com:RSE-Sheffield/collaborative_github_exercise.git (fetch)
+upstream	git@github.com:RSE-Sheffield/collaborative_github_exercise.git (push)
+```
+This tells us the names of the remote repositories we have configured `origin` and `upstream` and their URLs.
+
+
+### Pull upstream changes
+
+To pull the upstream changes from the **`RSE-Sheffield`** repository, run the following commands
+```
+$ git checkout master
+$ git fetch upstream
+$ git merge upstream/master
+```
+Git's default behaviour is to add these commits to the master branch in what is known as a "fast-forward merge". Find out more about [fast-forward git merging](https://ariya.io/2013/09/fast-forward-git-merge).
+
+
+
+### Push changes to your **origin** remote
+
+The **origin** remote is **your fork of the repository** that's labelled with your GitHub profile avatar.
+
+If we run:
+```
+$ git fetch origin
+# followed by
+$ git status
+```
+We can see, **it is now lagging behind** the upstream remote. That's because it's still missing all the files contributed by the other participants.
+
+To synchronise your **origin** remote, **Push**  the changes you've just pulled into your local repository by running:
```

Comment:
I really hate the use of the word _synchronise_ in the context of git, because I have no idea what it means, but I accept that everyone else uses it.

"pulled" however should definitely be replaced with "fetched"